api/v1alpha1/server_types.go

```diff
@@ -55,6 +55,18 @@ type BMCAccess struct {
 	BMCSecretRef v1.LocalObjectReference `json:"bmcSecretRef"`
 }
 
+type BootOrder struct {
+	Name     string `json:"name"`
+	Priority int    `json:"priority"`
+	Device   string `json:"device"`
+}
+
+type BIOSSettings struct {
+	Version string `json:"version"`
+	//maybe use map[string]intstr.IntOrString?!; interface not possible
```

Comment:
I am not sure that this will be serializable and if we can generate a proper OpenAPI spec out of this. We can check if kube is using `IntOrString` somewhere in their types.

Reply:
I think we may need to use rawjson here somehow, if we want to support all types. Dell for example also has type map or enumeration:

```
"Type": "Enumeration",
"Value": [
{
"ValueDisplayName": "Enabled",
"ValueName": "Enabled"
},
{
"ValueDisplayName": "Disabled",
"ValueName": "Disabled"
}
],
```

Reply:
Do we need then some kind of a `type` field in the `BIOSSettings` so we can cast it into a corresponding vendor specific struct?

Reply:
I was thinking about that. It would make it more complicated for the user though. Should we maybe just start with strings? so we would support numbers and strings for now.

Currently I am casting strings to int, if the type is an Integer according to the registry

Reply:
So are we ok to use `map[string]string` for now?